### DISCUSSION TITLE
bmp软解无效

### DISCUSSION THREAD

**Discussion by zhengcling** [Tencent/libpag#2214] [Q&A]
OPPO 1加手机微信浏览器，注册了ffavc软解之后，仍然报错“用户交互之后才可播放”。并且其他非BMP的文件也无法正常播放了。
iPhone 14 pro Safari 省电模式，注册了ffavc软解之后，无报错，但是仍然不能直接播放

请问这种情况如何避免呢?

```
        const pagManager = await window.libpag.PAGInit({
          locateFile: () => new URL('./assets/libpag.wasm', import.meta.url).href,
        });
        const FFAVC = await window.ffavc.FFAVCInit({
          locateFile: () => new URL('./assets/ffavc.wasm', import.meta.url).href,
        });
        const ffavcDecoderFactory = new FFAVC.FFAVCDecoderFactory();
        pagManager.registerSoftwareDecoderFactory(ffavcDecoderFactory);
        ...
        await pagManager.PAGView.init(pagFile, canvas)
```

libpag v4.3.47
ffavc v1.0.2

**Comment by kevingpqi123** [comment]
这里是浏览器的机制，和软解没啥关系，需要用户触发才能播放，我们也没有太多的办法，从素材层面而言，避免使用 BMP 预合成或 BMP 导出成位图序列（此时文件会偏大）可以解决这个问题

**Reply by zhengcling** [reply]
![无标题](https://github.com/Tencent/libpag/assets/27359061/f7b22752-9d72-4396-8060-55e620f13c0d)

是我理解错误了？这不是写的软解可以解决这个问题？

**Reply by kevingpqi123** [reply]
如果是这种情况，应该是软解没有注册成功，建议检查下软解注册

**Comment by pecliu** [comment]
类似的问题，注册软解后依然无法自动播放。
从 console 的日志来看已经走到了 ffavc 里面

**Comment by CodeJhF** [comment]
这条我们在当时的回复里已经给出结论，这里就不重复贴了，先关闭本条。
如果按当时的结论排查后仍有问题，欢迎回复补充最小复现 demo 与版本号（4.4 / 4.5），我们会重新看。（stale-close）
